project/leo27945875/README.rst

```diff
@@ -0,0 +1,112 @@
+========
+Proposal
+========
+
+Basic Information
+=================
+
+* Project: pyGMRES 
+* GitHub: https://github.com/leo27945875/pyGMRES
+
+Problem to Solve
+================
+
+The **Generalized-Minimal-Residual** (GMRES) solves a *large scale linear 
+system Ax = b* using efficient iterative method. This project will be 
+implemented in both dense and sparse matrix versions.
+
+Prospective Users
+=================
+
+GMRES can be used in many kinds of applications, as long as the application 
+needs to solve large scale linear systems. For instance: construction 
+engineering (structural analysis), EDA tools (IC design verification), ...etc.
+
+System Architecture
+===================
+
+This project will provide Python API and C++ implementation with pybind11 to 
+connect Python and C++. If time allows, there will be a multi-threading version 
+for matrix multiplication with OpenMP.
+
+Input: 
+   A matrix A and a vector b in dense matrices (numpy array) or sparse matrices 
+   (scipy CSR sparse matrix)
+  
+Output: 
+  A vector x in dense matrix (numpy array) or sparse matrix (scipy CSR sparse 
+  matrix)
+
+API Description
+===============
+
+**For dense matrices:**
+
+.. code-block:: python
+    
+    # Solve a large-scale linear system Ax = b.
+    
+    import pyGMRES
+    import numpy as np
+    
+    mat_A: np.ndarray = ...
+    vec_b: np.ndarray = ...
+    vec_x: np.ndarray = pyGMRES.solve_dense(mat_A, vec_b)
+    
+    
+**For sparse matrices:**
```

Comment:
Please specify the data structure for your sparse matrix in C++.

Reply:
Sparse matrix calls for significant amount of code development.  If you are not confident about your perseverance with C++, please consider to remove it from the scope of the term project.

I'd be interested in learning why do you propose to work on sparse matrix solution.

Reply:
I have specified the C++ data structure for the sparse matrix in README.

I want to implement the sparse version of GMRES because I saw [this](https://www.104.com.tw/job/8dg3i?jobsource=google). Furthermore, many finite element method applications require solving large-scale sparse linear systems, so I think it is valuable to implement the sparse version of GMRES.

Reply:
You may need a reference implementation with dense matrix to validate what you do with sparse matrix. This is one reason why sparse matrix optimization is hard.